File: Chapter3/R/Example5.R
# Chapter 3 - Binomial Distribution
# Example 5: Comparing B(20, p) for p = 0.2, 0.5, 0.8

# Parameters
n     <- 20
probs <- c(0.2, 0.5, 0.8)
x     <- 0:n

# Individual plots + comparison
par(mfrow = c(2, 2))

for (p in probs) {
  pmf      <- dbinom(x, n, p)
  mean_val <- n * p
  sd_val   <- sqrt(n * p * (1 - p))
  barplot(pmf, names.arg = x, col = 'steelblue',
          xlab   = 'k', ylab = 'P(X = k)',
          main   = sprintf('B(%d, %.1f): E(X)=%.1f, SD=%.2f', n, p, mean_val, sd_val),
          border = 'black')
}

# Overlay comparison
plot(x, dbinom(x, n, probs[1]), type = 'b', pch = 19, col = 'blue',
     xlab = 'k', ylab = 'P(X = k)', main = 'Distribution Comparison',
     ylim = c(0, 0.25), lwd = 2)
lines(x, dbinom(x, n, probs[2]), type = 'b', pch = 19, col = 'red',   lwd = 2)
lines(x, dbinom(x, n, probs[3]), type = 'b', pch = 19, col = 'green', lwd = 2)
legend('topright',
       legend = paste('p =', probs),
       col    = c('blue', 'red', 'green'), lwd = 2, pch = 19)

par(mfrow = c(1, 1))

File: Chapter3/Python/Example5.py
# Chapter 3 - Binomial Distribution
# Example 5: Comparing B(20, p) for p = 0.2, 0.5, 0.8

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import binom

# Parameters
n     = 20
probs = [0.2, 0.5, 0.8]
x     = np.arange(0, n + 1)

# Individual plots + comparison
plt.figure(figsize=(12, 8))

for i, p in enumerate(probs, 1):
    plt.subplot(2, 2, i)
    pmf  = binom.pmf(x, n, p)
    mean = n * p
    sd   = np.sqrt(n * p * (1 - p))
    plt.bar(x, pmf, color=f'C{i-1}', alpha=0.7, edgecolor='black')
    plt.xlabel('k')
    plt.ylabel('P(X = k)')
    plt.title(f'B({n}, {p}): E(X)={mean:.1f}, SD={sd:.2f}')
    plt.grid(axis='y', alpha=0.3)
    plt.xticks(range(0, n + 1, 2))

# Overlay comparison
plt.subplot(2, 2, 4)
for i, p in enumerate(probs):
    pmf = binom.pmf(x, n, p)
    plt.plot(x, pmf, 'o-', label=f'p = {p}', linewidth=2)

plt.xlabel('k')
plt.ylabel('P(X = k)')
plt.title('Distribution Comparison')
plt.legend()
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()
